wallet/src/main/java/com/android/identity/issuance/WalletDocumentMetadata.kt

```diff
@@ -0,0 +1,167 @@
+package com.android.identity.issuance
+
+import com.android.identity.cbor.annotation.CborSerializable
+import com.android.identity.document.DocumentMetadata
+import com.android.identity.issuance.remote.WalletServerProvider
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.io.bytestring.ByteString
+import kotlinx.io.bytestring.isEmpty
+import kotlin.concurrent.Volatile
+
+/**
+ * [DocumentMetadata] implementation for the Wallet app.
+ */
+class WalletDocumentMetadata private constructor(
+    serializedData: ByteString?,
+    private val saveFn: suspend (data: ByteString) -> Unit
+) : DocumentMetadata {
+    private val lock = Mutex()
+    private val data: Data = if (serializedData == null || serializedData.isEmpty()) {
+        Data()
+    } else {
+        Data.fromCbor(serializedData.toByteArray())
+    }
+
+    override val provisioned get() = data.provisioned
+
+    suspend fun markAsProvisioned() = lock.withLock {
+        check(!data.provisioned)
+        data.provisioned = true
+    }
+
+    override val displayName get() = data.documentConfiguration!!.displayName
+    override val typeDisplayName get() = data.documentConfiguration!!.typeDisplayName
+    override val cardArt get() = ByteString(data.documentConfiguration!!.cardArt)
+    override val issuerLogo get() = ByteString(data.issuingAuthorityConfiguration!!.issuingAuthorityLogo)
+
+    /** The identifier for the [IssuingAuthority] the credential belongs to */
+    val issuingAuthorityIdentifier: String get() = data.issuingAuthorityIdentifier!!
+
+    /**
+     * The identifier for the credential, as assigned by the issuer
```

Comment:
Should this be named credentialIdentifier? Or should this docstring refer to it as the identifier for the document, rather than the credential? If the document has multiple credentials, what does this field mean?

Or is this accurate, and the different APIs use different terminology?